src/main/java/com/microsoft/sqlserver/jdbc/Parameter.java

```diff
@@ -625,24 +625,18 @@ private void setTypeDefinition(DTV dtv) {
                          * generic type info can be used as before.
                          */
                         if (userProvidesScale) {
-                            param.typeDefinition = con
-                                    .isKatmaiOrLater() ? (SSType.DATETIME2.toString() + "(" + outScale + ")")
-                                                       : (SSType.DATETIME.toString());
+                            param.typeDefinition = getDatetimeDataType(con, outScale);
```

Comment:
this check still need to be there for !isKatmaiOrLater()as DATATIME2 is not supported before

same with line 632

Reply:
perhaps we should log an error if it's set to datatime2 and pre 2008